pkg/apis/core/v1alpha1/federatedcluster_types.go

```diff
@@ -25,35 +25,22 @@ import (
 
 // FederatedClusterSpec defines the desired state of FederatedCluster
 type FederatedClusterSpec struct {
-	// Name of the cluster registry Cluster resource from which to source api
-	// endpoints.
-	// TODO(marun) should this go away in favor of a 1:1 mapping?
-	ClusterRef LocalClusterReference `json:"clusterRef,omitempty"`
+	// The API endpoint of the member cluster. This can be a hostname,
+	// hostname:port, IP or IP:port.
+	APIEndpoint string `json:"apiEndpoint"`
```

Comment:
While I'm curious to know why the original calculation involving the CIDR was present, I think this is sufficient for now. We can re-add it in the future if there's a need.

Reply:
LGTM otherwise.

Reply:
Seems like we need to add an optional CA bundle here - do we not?

Reply:
CA is probably embedded in the secret which is referenced by this object (just few lines below).

Reply:
> CA is probably embedded in the secret which is referenced by this object (just few lines below).

Yeah, I was thinking that after I made the above comment. So, I think that implies that to join a cluster that requires you to configure a CA bundle that the kubeconfig file that you use to join the cluster has to contain the CA bundle information in the relevant context - is that accurate?

Reply:
That is correct - the CA bundle is stored in the secret.  As per the follow-on PR that cleans up secret handling, the code that builds the kubeconfig for accessing the member cluster sources values for `token` and `ca.crt` keys from the secret.  I'm not sure that the CA bundle needs to be stored in the secret, though, given that it's effectively a public key.  That's just the way it's always been done since fedv1.